// mindee/src/product/eu/driverLicense/driverLicenseV1Document.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DriverLicenseV1Document = void 0;
const common_1 = require("../../../parsing/common");
const standard_1 = require("../../../parsing/standard");
/**
 * Driver License API version 1.0 document data.
 */
class DriverLicenseV1Document {
    constructor(rawPrediction, pageId) {
        this.address = new standard_1.StringField({
            prediction: rawPrediction["address"],
            pageId: pageId,
        });
        this.category = new standard_1.StringField({
            prediction: rawPrediction["category"],
            pageId: pageId,
        });
        this.countryCode = new standard_1.StringField({
            prediction: rawPrediction["country_code"],
            pageId: pageId,
        });
        this.dateOfBirth = new standard_1.DateField({
            prediction: rawPrediction["date_of_birth"],
            pageId: pageId,
        });
        this.documentId = new standard_1.StringField({
            prediction: rawPrediction["document_id"],
            pageId: pageId,
        });
        this.expiryDate = new standard_1.DateField({
            prediction: rawPrediction["expiry_date"],
            pageId: pageId,
        });
        this.firstName = new standard_1.StringField({
            prediction: rawPrediction["first_name"],
            pageId: pageId,
        });
        this.issueAuthority = new standard_1.StringField({
            prediction: rawPrediction["issue_authority"],
            pageId: pageId,
        });
        this.issueDate = new standard_1.DateField({
            prediction: rawPrediction["issue_date"],
            pageId: pageId,
        });
        this.lastName = new standard_1.StringField({
            prediction: rawPrediction["last_name"],
            pageId: pageId,
        });
        this.mrz = new standard_1.StringField({
            prediction: rawPrediction["mrz"],
            pageId: pageId,
        });
        this.placeOfBirth = new standard_1.StringField({
            prediction: rawPrediction["place_of_birth"],
            pageId: pageId,
        });
    }
    /**
     * Default string representation.
     */
    toString() {
        const outStr = `:Country Code: ${this.countryCode}
:Document ID: ${this.documentId}
:Driver License Category: ${this.category}
:Last Name: ${this.lastName}
:First Name: ${this.firstName}
:Date Of Birth: ${this.dateOfBirth}
:Place Of Birth: ${this.placeOfBirth}
:Expiry Date: ${this.expiryDate}
:Issue Date: ${this.issueDate}
:Issue Authority: ${this.issueAuthority}
:MRZ: ${this.mrz}
:Address: ${this.address}`.trimEnd();
        return (0, common_1.cleanOutString)(outStr);
    }
}
exports.DriverLicenseV1Document = DriverLicenseV1Document;
