// mindee/src/product/internationalId/internationalIdV1Document.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InternationalIdV1Document = void 0;
const common_1 = require("../../parsing/common");
const standard_1 = require("../../parsing/standard");
/**
 * Document data for International ID, API version 1.
 */
class InternationalIdV1Document {
    constructor(rawPrediction, pageId) {
        /** The first names or given names of the document holder. */
        this.givenNames = [];
        /** The surnames of the document holder. */
        this.surnames = [];
        this.address = new standard_1.StringField({
            prediction: rawPrediction["address"],
            pageId: pageId,
        });
        this.birthDate = new standard_1.DateField({
            prediction: rawPrediction["birth_date"],
            pageId: pageId,
        });
        this.birthPlace = new standard_1.StringField({
            prediction: rawPrediction["birth_place"],
            pageId: pageId,
        });
        this.countryOfIssue = new standard_1.StringField({
            prediction: rawPrediction["country_of_issue"],
            pageId: pageId,
        });
        this.documentNumber = new standard_1.StringField({
            prediction: rawPrediction["document_number"],
            pageId: pageId,
        });
        this.documentType = new standard_1.ClassificationField({
            prediction: rawPrediction["document_type"],
        });
        this.expiryDate = new standard_1.DateField({
            prediction: rawPrediction["expiry_date"],
            pageId: pageId,
        });
        rawPrediction["given_names"] &&
            rawPrediction["given_names"].map((itemPrediction) => this.givenNames.push(new standard_1.StringField({
                prediction: itemPrediction,
                pageId: pageId,
            })));
        this.issueDate = new standard_1.DateField({
            prediction: rawPrediction["issue_date"],
            pageId: pageId,
        });
        this.mrz1 = new standard_1.StringField({
            prediction: rawPrediction["mrz1"],
            pageId: pageId,
        });
        this.mrz2 = new standard_1.StringField({
            prediction: rawPrediction["mrz2"],
            pageId: pageId,
        });
        this.mrz3 = new standard_1.StringField({
            prediction: rawPrediction["mrz3"],
            pageId: pageId,
        });
        this.nationality = new standard_1.StringField({
            prediction: rawPrediction["nationality"],
            pageId: pageId,
        });
        this.sex = new standard_1.StringField({
            prediction: rawPrediction["sex"],
            pageId: pageId,
        });
        rawPrediction["surnames"] &&
            rawPrediction["surnames"].map((itemPrediction) => this.surnames.push(new standard_1.StringField({
                prediction: itemPrediction,
                pageId: pageId,
            })));
    }
    /**
     * Default string representation.
     */
    toString() {
        const surnames = this.surnames.join("\n           ");
        const givenNames = this.givenNames.join("\n              ");
        const outStr = `:Document Type: ${this.documentType}
:Document Number: ${this.documentNumber}
:Country of Issue: ${this.countryOfIssue}
:Surnames: ${surnames}
:Given Names: ${givenNames}
:Gender: ${this.sex}
:Birth date: ${this.birthDate}
:Birth Place: ${this.birthPlace}
:Nationality: ${this.nationality}
:Issue Date: ${this.issueDate}
:Expiry Date: ${this.expiryDate}
:Address: ${this.address}
:Machine Readable Zone Line 1: ${this.mrz1}
:Machine Readable Zone Line 2: ${this.mrz2}
:Machine Readable Zone Line 3: ${this.mrz3}`.trimEnd();
        return (0, common_1.cleanOutString)(outStr);
    }
}
exports.InternationalIdV1Document = InternationalIdV1Document;
